### DISCUSSION TITLE
Cannot Enable Github Pages via API: Blank 500 Error

### DISCUSSION THREAD

**Discussion by hodgef** [community/community#24343] [Apps, API and Webhooks]
<p>Hello, I’m starting to get blank 500 “Internal Server Error” responses when I try to enable Github Pages through the API. This used to work a couple days ago.</p>
<p>Request URL: <a href="https://api.github.com/repos/hodgef/my-new-site-2/pages" rel="nofollow noopener">https://api.github.com/repos/hodgef/my-new-site-2/pages</a><br>
Request Method: POST<br>
Status Code: 500 Internal Server Error</p>
<p>The weird thing is that the response is blank so I have no clue what’s causing it.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2b193aecaf478b9dbd48c32219e099e78b8f10bf.jpeg" data-download-href="/uploads/short-url/69gAmZQdtBmgrM3NLg7IcSjCdGv.jpeg?dl=1" title="Capture3" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/25509135/181084296-a13fc540-7b46-494a-a813-a28ce3597c8d.jpeg" alt="Capture3" data-base62-sha1="69gAmZQdtBmgrM3NLg7IcSjCdGv" width="690" height="342" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2b193aecaf478b9dbd48c32219e099e78b8f10bf_2_690x342.jpeg, https://aws1.discourse-cdn.com/github/optimized/2X/2/2b193aecaf478b9dbd48c32219e099e78b8f10bf_2_1035x513.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/2X/2/2b193aecaf478b9dbd48c32219e099e78b8f10bf.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2b193aecaf478b9dbd48c32219e099e78b8f10bf_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture3</span><span class="informations">1328×659 83.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I would appreciate any help, thanks!</p>

**Comment by hodgef** [comment]
<p>It seems this issue can also be reproduced via postman, please help</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/b7a4f6d1ef677c2a3ebcece8d4eafb8ba8bcf603.png" data-download-href="/uploads/short-url/qcAOhZVEUVZoZQFw9H22GQD1zxh.png?dl=1" title="err" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/25509135/181084299-415c6d1c-df8f-43c2-ad98-3014490a6391.png" alt="err" data-base62-sha1="qcAOhZVEUVZoZQFw9H22GQD1zxh" width="665" height="499" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/b7a4f6d1ef677c2a3ebcece8d4eafb8ba8bcf603_2_665x499.png, https://aws1.discourse-cdn.com/github/original/2X/b/b7a4f6d1ef677c2a3ebcece8d4eafb8ba8bcf603.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/b/b7a4f6d1ef677c2a3ebcece8d4eafb8ba8bcf603.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/b7a4f6d1ef677c2a3ebcece8d4eafb8ba8bcf603_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">err</span><span class="informations">926×696 24.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p><a href="https://docs.github.com/en/rest/reference/repos#create-a-github-pages-site" class="onebox" target="_blank" rel="nofollow noopener">https://docs.github.com/en/rest/reference/repos#create-a-github-pages-site</a></p>

**Comment by francisfuzz** [comment]
<p>:wave: hello there <a class="mention" href="/hodgef">@hodgef</a> and welcome to the GitHub Support Community!</p>
<p>We appreciate the report––that <code>500 Internal Server Error</code> is not expected behavior. I surfaced this in our logs, but I also noticed that neither <code>hodgef/my-site-test</code> nor <code>hodgef/my-new-site-2</code> exist as <a href="https://github.com/hodgef?tab=repositories">public repositories under your account</a>. If these are private repositories, could you try enabling GitHub Pages via the API again and sharing the full output of a <code>curl -v</code> request that indicates the same error? If they no longer exist, could you try reproducing this on a public repository and see if that helps?</p>

**Comment by betosg** [comment]
<p>I facing the same problem. I enabled the repository to public view.</p>
<pre><code class="lang-auto">curl -v --location --request POST 'https://api.github.com/repos/betosg/amppage.com.br/pages' \
--header 'Accept: application/vnd.github.switcheroo-preview+json' \
--header 'Authorization: Basic  XXXXXXXXXXXXXXXXXXX' \
--header 'Content-Type: text/plain' \
--data-raw '{
    "source": {
        "branch": "master",
        "path": "/"
  }
}'
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying 54.207.47.76:443...
* TCP_NODELAY set
* Connected to api.github.com (54.207.47.76) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=GitHub, Inc.; CN=*.github.com
*  start date: Jun 22 00:00:00 2020 GMT
*  expire date: Aug 17 12:00:00 2022 GMT
*  subjectAltName: host "api.github.com" matched cert's "*.github.com"
*  issuer: C=US; O=DigiCert Inc; OU=www.digicert.com; CN=DigiCert SHA2 High Assurance Server CA
*  SSL certificate verify ok.
&gt; POST /repos/betosg/amppage.com.br/pages HTTP/1.1
&gt; Host: api.github.com
&gt; User-Agent: curl/7.68.0
&gt; Accept: application/vnd.github.switcheroo-preview+json
&gt; Authorization: Basic XXXXXXXXXXXXXXXXXXX
&gt; Content-Type: text/plain
&gt; Content-Length: 71
&gt; 
* upload completely sent off: 71 out of 71 bytes
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Mark bundle as not supporting multiuse
&lt; HTTP/1.1 500 Internal Server Error
&lt; Date: Mon, 27 Jul 2020 20:17:28 GMT
&lt; Content-Type: application/json; charset=utf-8
&lt; Content-Length: 0
&lt; Server: GitHub.com
&lt; Status: 500 Internal Server Error
&lt; X-RateLimit-Limit: 5000
&lt; X-RateLimit-Remaining: 4991
&lt; X-RateLimit-Reset: 1595883308
&lt; X-GitHub-Media-Type: github.switcheroo-preview; format=json
&lt; Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset
&lt; Access-Control-Allow-Origin: *
&lt; Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
&lt; X-Frame-Options: deny
&lt; X-Content-Type-Options: nosniff
&lt; X-XSS-Protection: 1; mode=block
&lt; Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
&lt; Content-Security-Policy: default-src 'none'
&lt; Vary: Accept-Encoding, Accept, X-Requested-With
&lt; X-GitHub-Request-Id: B096:4C68:B5AE79:101BD58:5F1F3658
&lt; 
* Connection #0 to host api.github.com left intact
</code></pre>

**Comment by hodgef** [comment]
<p>Hello <a class="mention" href="/francisfuzz">@francisfuzz</a> , I just tried this with a new public repo. Here’s my curl -v output, same as Beto:</p>
<pre><code class="lang-auto">curl -v --location --request POST 'https://api.github.com/repos/hodgef/my-test-site-3/pages' --header 'Accept: applicat
ion/vnd.github.switcheroo-preview+json' --header 'Authorization: Basic XXX' --header 'Content-Type: text/p
lain' --data-raw '{
    "source": {
        "branch": "master",
        "path": "/"
  }
}'
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying 140.82.112.5...
* TCP_NODELAY set
* Connected to api.github.com (140.82.112.5) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=GitHub, Inc.; CN=*.github.com
*  start date: Jun 22 00:00:00 2020 GMT
*  expire date: Aug 17 12:00:00 2022 GMT
*  subjectAltName: host "api.github.com" matched cert's "*.github.com"
*  issuer: C=US; O=DigiCert Inc; OU=www.digicert.com; CN=DigiCert SHA2 High Assurance Server CA
*  SSL certificate verify ok.
&gt; POST /repos/hodgef/my-test-site-3/pages HTTP/1.1
&gt; Host: api.github.com
&gt; User-Agent: curl/7.58.0
&gt; Accept: application/vnd.github.switcheroo-preview+json
&gt; Authorization: Basic XXX
&gt; Content-Type: text/plain
&gt; Content-Length: 71
&gt;
* upload completely sent off: 71 out of 71 bytes
&lt; HTTP/1.1 500 Internal Server Error
&lt; Server: GitHub.com
&lt; Date: Mon, 27 Jul 2020 20:34:22 GMT
&lt; Content-Type: application/json; charset=utf-8
&lt; Content-Length: 0
&lt; Status: 500 Internal Server Error
&lt; X-RateLimit-Limit: 5000
&lt; X-RateLimit-Remaining: 4994
&lt; X-RateLimit-Reset: 1595885368
&lt; X-OAuth-Scopes: repo
&lt; X-Accepted-OAuth-Scopes:
&lt; X-OAuth-Client-Id: 548a6e02e01ae9015446
&lt; X-GitHub-Media-Type: github.switcheroo-preview; format=json
&lt; Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset
&lt; Access-Control-Allow-Origin: *
&lt; Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
&lt; X-Frame-Options: deny
&lt; X-Content-Type-Options: nosniff
&lt; X-XSS-Protection: 1; mode=block
&lt; Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
&lt; Content-Security-Policy: default-src 'none'
&lt; Vary: Accept-Encoding, Accept, X-Requested-With
&lt; X-GitHub-Request-Id: D657:5FD9:26AF47C:4255A90:5F1F3A4E
&lt;
* Connection #0 to host api.github.com left intact
</code></pre>
<p>Thanks for looking into this, I hope it’s fixable :slight_smile:</p>

**Comment by francisfuzz** [comment]
<p><a class="mention" href="/betosg">@betosg</a> &amp; <a class="mention" href="/hodgef">@hodgef</a> - Thank you so much for sharing these example <code>curl -v</code> requests with us. That’s extremely helpful for engineers to check out.</p>
<p>As a next step, we’ve escalated this to our engineering team and while we don’t have a timeline for when this will be resolved, we will follow up here when we have an update from them. :bowing_man:</p>

**Comment by yoannchaudet** [comment]
<p>Hello :wave:!</p>
<p>Pages team here. We just deployed a fix that should address this issue. Let us know if you still see 500 errors on POST.</p>
<p>Sorry for the inconvenience. We are preparing some upcoming changes for Pages and we let this regression slipped.</p>

**Comment by hodgef** [comment]
<p>It works! Thank you <a class="mention" href="/yoannchaudet">@yoannchaudet</a> &amp; team for fixing this so quickly.</p>

**Comment by betosg** [comment]
<p><a class="mention" href="/francisfuzz">@francisfuzz</a> <a class="mention" href="/yoannchaudet">@yoannchaudet</a><br>
Good morning, the problem happened again, it is showing error 500. Could you check again, please!</p>
<pre><code class="lang-auto">curl -v --location --request POST 'https://api.github.com/repos/betosg/melhoresofertasford.com.br/pages' \
--header 'Accept: application/vnd.github.switcheroo-preview+json' \
--header 'Authorization: token XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX' \
--header 'Content-Type: text/plain' \
--data-raw '{
    "source": {
        "branch": "master",
        "path": "/"
  }
}'
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying 140.82.114.5:443...
* TCP_NODELAY set
* Connected to api.github.com (140.82.114.5) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=GitHub, Inc.; CN=*.github.com
*  start date: Jun 22 00:00:00 2020 GMT
*  expire date: Aug 17 12:00:00 2022 GMT
*  subjectAltName: host "api.github.com" matched cert's "*.github.com"
*  issuer: C=US; O=DigiCert Inc; OU=www.digicert.com; CN=DigiCert SHA2 High Assurance Server CA
*  SSL certificate verify ok.
&gt; POST /repos/betosg/melhoresofertasford.com.br/pages HTTP/1.1
&gt; Host: api.github.com
&gt; User-Agent: curl/7.68.0
&gt; Accept: application/vnd.github.switcheroo-preview+json
&gt; Authorization: token XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
&gt; Content-Type: text/plain
&gt; Content-Length: 71
&gt; 
* upload completely sent off: 71 out of 71 bytes
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Mark bundle as not supporting multiuse
&lt; HTTP/1.1 500 Internal Server Error
&lt; Server: GitHub.com
&lt; Date: Mon, 07 Dec 2020 14:01:25 GMT
&lt; Content-Type: application/json; charset=utf-8
&lt; Content-Length: 0
&lt; Status: 500 Internal Server Error
&lt; X-OAuth-Scopes: admin:enterprise, admin:gpg_key, admin:org, admin:org_hook, admin:public_key, admin:repo_hook, delete:packages, delete_repo, gist, notifications, repo, user, workflow, write:discussion, write:packages
&lt; X-Accepted-OAuth-Scopes: 
&lt; X-GitHub-Media-Type: github.switcheroo-preview; format=json
&lt; X-RateLimit-Limit: 5000
&lt; X-RateLimit-Remaining: 4973
&lt; X-RateLimit-Reset: 1607349911
&lt; X-RateLimit-Used: 27
&lt; Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset
&lt; Access-Control-Allow-Origin: *
&lt; Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
&lt; X-Frame-Options: deny
&lt; X-Content-Type-Options: nosniff
&lt; X-XSS-Protection: 1; mode=block
&lt; Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
&lt; Content-Security-Policy: default-src 'none'
&lt; Vary: Accept-Encoding, Accept, X-Requested-With
&lt; X-GitHub-Request-Id: B57E:04A5:2D7DC02:4C3F551:5FCE35B4
&lt; 
* Connection #0 to host api.github.com left intact
</code></pre>

**Comment by mattlangan** [comment]
<p>I’m experiencing the same issue. A previously-working <code>POST</code> to <code>../username/repository_name/pages</code> to convert a repo to a Pages site is returning the same <code>500</code> response.</p>

**Comment by francisfuzz** [comment]
<p><a class="mention" href="/betosg">@betosg</a> - Thanks for sharing this instance with us. I’ve forwarded it to our engineering team for their awareness so they can investigate this further.</p>
